care/facility/api/serializers/patient_consultation.py

```diff
@@ -515,6 +515,22 @@ def validate_encounter_date(self, value):
             raise ValidationError(
                 {"encounter_date": "This field value cannot be in the future."}
             )
+
+        last_consultation = (
+            PatientConsultation.objects.filter(
+                patient=self.validated_data["patient"], discharge_date__isnull=False
+            )
+            .order_by("-encounter_date")
+            .first()
+        )
+        if last_consultation and value < last_consultation.encounter_date:
+            raise ValidationError(
+                {
+                    "encounter_date": [
+                        "This field value must be greater than the last discharge date."
+                    ]
+                }
+            )
```

Comment:
move this to validate() as it depends on multiple fields

Reply:
Done